### PR TITLE
Add s2i builder images task

### PR DESCRIPTION
**Pull request by piyush-garg**
Add tasks to build image for your source code
using s2i builder image

Runtime added
Java
PHP
Python
Ruby
nodejs
go
Perl
DotNet

Add scripts location to java tasks

### CODE REVIEW

**Comment by siamaksade** [comment]
Are all these tasks tested? I believe at least the s2i-java task doesn't work due to https://github.com/openshift/source-to-image/issues/973

As a workaround, the s2i scripts location specific to each s2i builder image should get passed to the `s2i` command via the `--image-scripts-url` argument.

**Comment by vdemeester** [comment]
> Are all these tasks tested? I believe at least the s2i-java task doesn't work due to [openshift/source-to-image#973](https://github.com/openshift/source-to-image/issues/973)
> 
> As a workaround, the s2i scripts location specific to each s2i builder image should get passed to the `s2i` command via the `--image-scripts-url` argument.

Indeed. We decide with @piyush-garg to open the PR "as is there was no bug", and then discuss some workaround. Technically it would also make it easy to revert the workaround once the bug is fixed :angel:

**Review by siamaksade — APPROVED** [review]

**Comment by siamaksade** [comment]
We should probably change to `quay.io/buildah/stable` image that the buildah team owns

File: s2i-nodejs/s2i-nodejs-task.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        default: "true"
+  outputs:
+    resources:
+      - name: image
+        type: image
+  steps:
+    - name: generate
+      image: quay.io/openshift-pipeline/s2i
+      workingdir: /workspace/source
+      command: ['s2i', 'build', '${inputs.params.PATH_CONTEXT}', 'centos/nodejs-${inputs.params.VERSION}-centos7', '--as-dockerfile', '/gen-source/Dockerfile.gen']
```
Comment (Ladicek) [inline s2i-nodejs/s2i-nodejs-task.yaml:28]: Did you talk to the RHOAR Node.js team about using their image(s)?

CC @lance @lholmquist

Comment (siamaksade) [reply s2i-nodejs/s2i-nodejs-task.yaml:28]: The Node.js images are used here like any other S2I use-cases. Are there any concerns?

**Review by siamaksade — APPROVED** [review]

File: s2i-nodejs/s2i-nodejs-task.yaml (hunk 1 above)
Comment (Ladicek) [reply s2i-nodejs/s2i-nodejs-task.yaml:28]: I don't have any concerns about how the images are used. I'm just wondering about _which_ images are used. I see that for Java, the images produced by the Red Hat OpenJDK team are used. Why not use the Node.js images produced by the RHOAR Node.js team? I believe they also have a community, CentOS-based variants, if that's a concern.

**Review by vdemeester — APPROVED** [review]
/lgtm

File: s2i-nodejs/s2i-nodejs-task.yaml (hunk 1 above)
Comment (siamaksade) [reply s2i-nodejs/s2i-nodejs-task.yaml:28]: Sure. That's the goal. It will be added #4

**Pull request merged**